modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue

```diff
@@ -135,22 +162,23 @@ export default {
     schema() {
       this.populateDocData();
     },
-    value: {
-      deep: true,
-      handler(newVal, oldVal) {
-        // The doc might be swapped out completely in cases such as the media
-        // library editor. Repopulate the fields if that happens.
-        if (
-          // If the fieldState had been cleared and there's new populated data
-          (!this.fieldState._id && newVal.data._id) ||
-          // or if there *is* active fieldState, but the new data is a new doc
-          (this.fieldState._id && newVal.data._id !== this.fieldState._id.data)
-        ) {
-          // repopulate the schema.
-          this.populateDocData();
-        }
+    'value.data._id'(_id) {
```

Comment:
This is all we were really doing with the deep watch before and it's more performant. I verified that switching docs in the media manager still correctly swaps what's in the schema view.